livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterpreter.java

```diff
@@ -205,6 +215,47 @@ public InterpreterResult interpret(String st, InterpreterContext context) {
   }
 
   @Override
+  public List<InterpreterCompletion> completion(String buf, int cursor,
+      InterpreterContext interpreterContext) {
+    List<InterpreterCompletion> candidates = Collections.emptyList();
+    try {
+      try {
+        candidates = callCompletion(new CompletionRequest(buf, getSessionKind(), cursor));
+      } catch (SessionNotFoundException e) {
+        LOGGER.warn("Livy session {} is expired, new session will be created.", sessionInfo.id);
```

Comment:
Have concern whether we need to recreate session when code completion is called.

Reply:
you are right. and the chance for getting a useful result is also probably quite low (since context is missing).
Will remove.
Just return empty result?